File: spoj/11-FCTRL/solution_11.py

def trailing_zeros(num:int):
    zeros = 0
    while num >= 5:
        num //= 5
        zeros += num
    return zeros

if __name__ == "__main__":
    n = int(input())
    inputs = [int(input()) for x in range(n)]
    output = map(trailing_zeros, inputs)
    [print(x) for x in list(output) ]

File: spoj/11-FCTRL/solution_11.cpp
#include <iostream>
#include <math.h>

using namespace std;

int trailingZeros(int num);

int main()
{
    int n;
    cin >> n;
    for (int i = 0; i < n; i++)
    {
        int num;
        cin >> num;
        cout << trailingZeros(num) << "\n";
    }
}

int trailingZeros(int num)
{
    int zeros = 0;
    int times = 1;
    int fives = pow(5, times);
    while (fives <= num)
    {
        zeros += num / fives;
        times++;
        fives = pow(5, times);
    }
    return zeros;
}
